apps/api/src/stripe/stripe.service.ts

```diff
@@ -220,19 +208,23 @@ export class StripeService {
     return customer
   }
 
-  async createProduct(campaignId: string, idempotencyKey: string): Promise<Stripe.Product> {
+  async createProduct(campaignId: string): Promise<Stripe.Product> {
     const campaign = await this.campaignService.getCampaignById(campaignId)
+    const idempotencyKey = crypto.randomUUID()
     if (!campaign) throw new Error(`Campaign with id ${campaignId} not found`)
 
     const productLookup = await this.stripeClient.products.search({
-      query: `-name:'${campaign.title}'`,
+      query: `metadata["campaignId"]:"${campaign.id}"`,
     })
 
-    if (productLookup) return productLookup.data[0]
+    if (productLookup.data.length) return productLookup.data[0]
     return await this.stripeClient.products.create(
       {
         name: campaign.title,
         description: `Donate to ${campaign.title}`,
+        metadata: {
+          campaignId: campaign.id,
```

Comment:
Do we need to do a migration for the existing products that were created without `campaign.id` ?